frontend/src/pages/analytics/traffic-source/modal/SourceCategoryModal.tsx

```diff
@@ -0,0 +1,364 @@
+        projectId: state.projectId,
+        appId: state.appId,
+        sourceCategory: category,
+      });
+      if (success) {
+        dispatch({ type: 'NewItem', category });
+        resetInput();
+        setSelectedItems([]);
+        setVisible(false);
+      }
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+
+  const actionEdit = async () => {
+    setLoading(true);
+    try {
+      const category: ISourceCategory = {
+        url: newDomain,
+        source: newSource,
+        category: selectedCategory?.value as ESourceCategory,
+        params: newPatterns.map((item) => item.label),
+      };
+      const { success }: ApiResponse<any> = await trafficSourceAction({
+        action: ITrafficSourceAction.UPDATE,
+        projectId: state.projectId,
+        appId: state.appId,
+        sourceCategory: category,
+      });
+      if (success) {
+        dispatch({ type: 'UpdateItem', category });
+        resetInput();
+        setSelectedItems([]);
+        setVisible(false);
+      }
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (selectedItems.length === 1) {
+      const selectedItem = selectedItems[0];
+      setNewDomain(selectedItem.url);
+      setNewSource(selectedItem.source);
+      setSelectedCategory(
+        categoryOptions.find((item) => item.value === selectedItem.category) ||
+          null
+      );
+      setNewPatterns(
+        selectedItem.params.map((item) => {
+          return { label: item };
+        })
+      );
+    }
+    if (modalType === t('analytics:metadata.trafficSource.modalType.new')) {
+      resetInput();
+    }
+  }, [selectedItems, modalType]);
+
+  return (
+    <>
+      <Modal
+        onDismiss={() => {
+          setVisible(false);
+        }}
+        visible={visible}
+        closeAriaLabel="Close modal"
+        footer={
+          <Box float="right">
+            <SpaceBetween direction="horizontal" size="xs">
+              <Button
+                onClick={() => {
+                  resetInput();
+                  setVisible(false);
+                }}
+                variant="link"
+              >
+                {t('button.cancel')}
+              </Button>
+              <Button
+                variant="primary"
+                loading={loading}
+                onClick={() => {
+                  if (checkInput()) {
+                    if (
+                      modalType ===
+                      t('analytics:metadata.trafficSource.modalType.edit')
+                    ) {
+                      actionEdit();
+                    } else {
+                      actionNew();
+                    }
+                  }
+                }}
+              >
+                {t('button.confirm')}
+              </Button>
+            </SpaceBetween>
+          </Box>
+        }
+        header={
+          defaultStr(
```

Comment:
change to  Template literals ([Template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals))